server/rooms.ts

```diff
@@ -1420,7 +1420,13 @@ export class GlobalRoomState {
 		let privateCrashMessage = null;
 
 		const upperStaffRoom = Rooms.get('upperstaff');
-		if (stack.includes("private")) {
+		let hasPrivateTerm = stack.includes('private');
+
+		const privateTerms = (Config.privatecrashterms ? [...Config.privatecrashterms] : []).map(item => new RegExp(item, "i"));
+		for (const term of privateTerms) {
+			if (term.test(stack) || term.test(err.message)) hasPrivateTerm = true;
```

Comment:
```suggestion
			if (typeof term === 'string' ? err.stack.includes(term) : term.test(err.stack)) {
				hasPrivateTerm = true;
				break;
			}
```

This will support both string matching and regex matching.